### PR TITLE
add float16 docs and tweak typehints

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: src/axolotl/utils/models.py [hunk 1]
```diff
@@ -70,7 +71,7 @@ def load_tokenizer(
 def load_model(
     base_model, base_model_config, model_type, tokenizer, cfg, adapter="lora"
 ):
-    # type: (str, str, str, AutoTokenizer, DictDefault, Optional[str]) -> Tuple[PreTrainedModel, Optional[PeftConfig]]
+    # type: (str, str, str, PreTrainedTokenizerBase, DictDefault, Optional[str]) -> Tuple[PreTrainedModel, Optional[PeftConfig]]
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py:75]: Ah, is this for the Fast tokenizer?

File: README.md [hunk 2]
```diff
@@ -264,6 +264,8 @@ See sample configs in [configs](configs) folder or [examples](examples) for quic
   bf16: true # require >=ampere
   fp16: true
   tf32: true # require >=ampere
+  bfloat16: true # require >=ampere, use instead of bf16 when you don't want AMP
+  float16: true # use instead of fp16 when you don't want AMP
```
Comment (NanoCode012) [inline README.md:268]: Is there a gpu limitation for this? `float16`

**Review by NanoCode012 — APPROVED** [review]
Looks good

File: README.md (hunk 2 above)
Comment (utensil) [reply README.md:268]: What does "when you don't want AMP" mean?

Comment (NanoCode012) [reply README.md:268]: @utensil , AMP stands for automatic mixed precision.

Comment (utensil) [reply README.md:268]: Thanks for the clarification. I mean it's not mentioned elsewhere so might cause confusion, also the interaction of AMP and the float types chosen here is not so clear.

Comment (NanoCode012) [reply README.md:268]: I agree the naming is unclear.

File: src/axolotl/utils/models.py (hunk 1 above)
Comment (winglian) [reply src/axolotl/utils/models.py:75]: no particular one, I just noticed the IDE wasn't properly suggesting the correct things.

File: README.md (hunk 2 above)
Comment (winglian) [reply README.md:268]: I added some additional copy, hopefully it helps

**Pull request merged**